### ISSUE TITLE
Out of stock warning when product is added to cart

### ISSUE THREAD

**Issue by Murat-06** [PrestaShop/PrestaShop#28523]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Is your feature request related to a problem?

I am using version 1.7.8.6 and the products I sell are generally used 1 stocked products. When the customer adds the product to the cart, the product is out of stock, it resets the product stock quantity, and when you add all the products in stock, it gives the same error.
 And when the customer sees the product out of stock label, he leaves the site because there is no product. Prestashop 1.6 does not give this warning in the same operand. When you add to the cart, the number of products does not decrease and does not give out of stock warning. When I check other platforms, it doesn't give an out of stock warning because your product has been added to the cart. I hope this issue can be fixed with a new update to version 1.7.8.6 now I'm sending you the screenshots Thanks

screenshots for prestashop 1.7

![image](https://user-images.githubusercontent.com/105777131/169006624-15a76f4a-7571-466e-abf4-7045dde7752d.png)
![image](https://user-images.githubusercontent.com/105777131/169006847-7bbaae77-ed43-432a-94b2-4dc02c3dfd8f.png)

Just added to the cart, even though the order has not been placed, the product's stock quantity has been reset and it is out of stock warning

When the last product is added to the cart, it gives out of stock warning.
and the customer leaves the site

### Describe the solution you'd like
> Edit by @Hlavtox 

**However, I also find this behavior confusing and would prefer to disable it.**
- Sometimes you call with a customer: 
  - You tell him that it's in stock.
  - He says: "It's not in stock!".
  - You say: "Don't you have it in cart?"
  - Him: "Oooh yes I do...."
- Or me as a merchant I am browsing my store and I wonder why the hell this product shows up when I select only "In stock" items in faceted search. Oh yes, I have it in cart.

@MatShir Do you think we can add an option to Prestashop to disable these virtual reservations and just show the real quantity?

Ping @kpodemski

**Comment by Hlavtox** [comment]
@Murat-06 Hello, this is just the behavior present on 1.7 version. When you add some product to cart, the shop substracts this quantity from the amount in stock. So if you have 2 items in stock and you add 1 piece to cart, the product page will show that 1 is in stock. (Because 1 is in cart.)

Prestashop 1.6 didn't do this, yes.

**However, I also find this behavior confusing and would prefer to disable it.**
- Sometimes you call with a customer: 
  - You tell him that it's in stock.
  - He says: "It's not in stock!".
  - You say: "Don't you have it in cart?"
  - Him: "Oooh yes I do...."
- Or me as a merchant I am browsing my store and I wonder why the hell this product shows up when I select only "In stock" items in faceted search. Oh yes, I have it in cart.

@MatShir Do you think we can add an option to Prestashop to disable these virtual reservations and just show the real quantity?

Ping @kpodemski

**Comment by Murat-06** [comment]
Of course, it is not possible for us to do this by calling the customer, because customers can shop from the site while you are sleeping.

Also, when it is added to the cart, there is no problem that it drops from the stock level. However, when it is out of stock, adding an Added to Cart warning instead of Out of Stock may make this system more usable, thank you
![image](https://user-images.githubusercontent.com/105777131/169026966-985a254d-ab4d-4dca-b18c-c9b5c0d4e0ab.png)

**Comment by Hlavtox** [comment]
We are currently remaking this and the availability will look much better on Hummingbird theme.

I think we can maybe improve this by showing something like
`You have 2 pieces of this product in cart`
on top of the page.

**Comment by Murat-06** [comment]
I'm looking forward to it thank you

**Comment by hibatallahAouadni** [comment]
Hello @Murat-06 

Thanks for pointing to this issue :pray: 
It seems like a feature request more than a bug, so I modified the issue description :relaxed: 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

PS: @Hlavtox thanks for your quick replies :superhero: 

Thank you

**Comment by melikslab** [comment]
Hi everyone! I might be a few years late, but I’ve created a small solution to this issue. In my case, I added a warning inside the product page to inform users that they already have the product in their cart.

I edited the file:
/themes/customtheme/templates/catalog/_partials/product-add-to-cart.tpl

and added this code:

```
{assign var='context' value=Context::getContext()}
{assign var='cart' value=$context->cart}

{block name='product_cart_message'}
  {if $cart && $cart->getProducts()}
    {foreach from=$cart->getProducts() item=cart_product}
      {if $cart_product.id_product == $product.id_product}
        <div class="product-cart-message alert alert-danger" style="text-align: center;">
          {l s='You already have %quantity% in your cart.' d='Shop.Theme.Actions' sprintf=['%quantity%' => $cart_product.cart_quantity]}
        </div>
        {break}
      {/if}
    {/foreach}
  {/if}
{/block}
```

This code checks if the product is already in the cart, and if it is, it displays a message on the product page. Hope it helps!

**Comment by boo-code** [comment]
The concrete proposal in this thread, showing the customer how many of the product they already hold, is not implemented on 9.2.x, but it is closer than it looks.

The value is already computed server side. `ProductController::getTemplateVarProduct()` sets it before presenting the product:

```php
$product['cart_quantity'] = $this->context->cart->getProductQuantity((int) $this->product->id, $product['id_product_attribute'])['quantity'];
```

and `ProductLazyArray::getQuantityRequired()` already consumes it, to stop asking for a minimum quantity the customer has partly satisfied. So the number behind "you have N of these in your cart" is available at presentation time; nothing renders it.

Checked on a 9.2 shop with the default theme: the product page carries `class="product__availability js-product-availability"` and no message about the cart anywhere. Hummingbird's templates contain no "in cart" wording outside `checkout/_partials/cart-detailed.tpl`, which is the cart page itself.

That makes this a theme-side change on top of data core already has, rather than the plumbing job it would have been in 2022, which is worth knowing for whoever picks up the availability rework mentioned above.
